docs/development/pigs/pig-010.rst

```diff
@@ -0,0 +1,304 @@
+boxes that should partially be removed, partially refactored to use
+``astropy-regions``. This PR lists the work to be done to get region-related
+code in shape for Gammapy.
+
+The scope of this PIG is small and limited to spatial sky and pixel regions. The
+question of more general data subspace selections that include energy, time or
+phase regions and selections, or general n-dimensional regions, or whether to
+introduce a field of view (FOV) coordinate frame and special FOV regions is not
+addressed here.
+
+Introduction
+============
+
+Pixel and sky regions. See section below: `Regions`_
+
+Spatial regions are used a lot in gamma-ray astronomy (and other domains of
+astronomy as well). Often a users chooses a region of interest, and then selects
+the observations or events or pixels for analysis. There are a few use cases to
+create regions within Gammapy, e.g. to restrict analysis to a maximum field of
+view offset, or to compute "off" regions for background estimation for a given
+"on" region.
+
+However, the places where regions appear in the Gammapy code and API are very
+few and overall region handling is pretty simple.
+
+The reason for this is that Gammapy is mostly a binned analysis framework, so
+mainly regions are used to create pixel masks, and then the analysis is
+mask-based, not involving region objects any more. See section below: `Masks`_.
+
+The second place where regions appear is to select objects of interest (events,
+observations, sources) for a given region. See section below: `Tables`_.
+
+Finally, reflected regions and PSF correction in 1D spectral analysis.
+See section below: `Rotated regions`_
+
+See section below: `Region strings`_
+
+Proposal
+========
+
+This section contains a detailed proposal concerning region-related code in
+Gammapy. The `Task list`_ at the end of this PIG references these descriptions.
+
+Regions
+-------
+
+One key feature that's currently still missing in ``astropy-regions`` is the
+implementation of the ``contains`` method of `regions.PolygonSkyRegion`_.
+
+This requires a spherical point in polygon algorithm, which probably is hard to
+implement. Discussion on this point is in `astropy-regions GH 46`_.
+
+- Implement ``regions.CircleSkyRegion.contains`` (see above)
+- Implement ``regions.PolygonSkyRegion.contains``, also use for rectangle? (see TODO)
+
+Merge `astropy-regions GH 221` to make it easier to plot sky regions? Is it a good idea or will it cause problems with "real" sky regions?
+
+Masks
+-----
+
+, via the existing
+`gammapy.maps.WcsGeom.region_mask`_ method, and then only the pixel mask is
+used.
+
+- Use bool or int as `dtype`?
+- Define meaning of 0 and 1 (probably use as-is)
+
+Sometimes binary masks are not enough. E.g. in ``scikit-image`` or
+``scipy.ndimage`` often "label images" are used, in cases where there are more
+than two disjoint regions ("in" and "out" as in a binary mask), e.g. 
+
+Image analysis
```

Comment:
@adonath and @registerrier - Thoughts on what to keep & polish, and what to remove in ``gammapy.image``?

The scope of the PIG isn't very large, so I thought we could discuss and decide here?

But if you prefer, I can also remove this again, and we do it later and somewhere else.

Reply:
I think describing what happens with `gammapy.image` is not focus of this PIG. I would maybe just add on sentence here, that region handling for any kind of image analysis is supported via `mask` attributes, which can be simply created with `geom.region_mask()`.